packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts

```diff
@@ -227,7 +223,10 @@ export default util.createRule<Options, MessageIds>({
 
         context.report({
           fix: function* (fixer) {
```

Comment:
after further investigation we could simplify this fixer to

```ts
          fix: function* (fixer) {
            const isUnaryNegation =
              node.parent != null && nodeIsUnaryNegation(node.parent);

            const mutatedNode = isUnaryNegation ? node.parent! : node;

            yield fixer.replaceText(
              mutatedNode,
              sourceCode.getText(comparison.expression),
            );

            const shouldNegate = comparison.literalBooleanInComparison
              ? !comparison.negated
              : comparison.negated;

            if (isUnaryNegation === shouldNegate) {
              yield fixer.insertTextBefore(mutatedNode, '!');

              if (!util.isStrongPrecedenceNode(comparison.expression)) {
                yield fixer.insertTextBefore(mutatedNode, '(');
                yield fixer.insertTextAfter(mutatedNode, ')');
              }
            }

            // if the expression `exp` is nullable, and we're not comparing to `true`,
            // we can just replace the entire comparison with `exp` or `!exp`
            if (
              comparison.expressionIsNullableBoolean &&
              !comparison.literalBooleanInComparison
            ) {
              // provide the default `true`
              yield fixer.insertTextBefore(mutatedNode, '(');
              yield fixer.insertTextAfter(mutatedNode, ' ?? true)');
            }
          }
```

that should allow us to provide better fixer for

```ts
## source
if (!(varBoolean === false)) {}
## current
if (!(!varBoolean)) {}
## new
if (varBoolean) {}
```

i'm not sure if we should push this with this fix